example/lib/main.dart

```diff
@@ -60,6 +69,58 @@ class _MyAppState extends State<MyApp> {
                       },
                     ),
                   ),
+                  // add a switch btn to set private or public video
+                  Padding(
+                    padding: const EdgeInsets.only(top: 3),
+                    child: LiteRollingSwitch(
+                      value: _isPublicVideo,
+                      width: 150,
+                      textOn: 'Public',
+                      textOff: 'Private',
+                      colorOn: Colors.blueGrey,
+                      colorOff: Colors.orangeAccent,
+                      iconOn: Icons.public,
+                      iconOff: Icons.lock,
+                      animationDuration: const Duration(milliseconds: 300),
+                      onChanged: (bool state) {
+                        _textEditingController.text =
+                            state ? _publicVideoId : _privateVideoId;
+                        setState(() {
+                          _isPublicVideo = state;
+                        });
+                      },
+                      onDoubleTap: () {},
+                      onSwipe: () {},
+                      onTap: () {},
+                    ),
+                  ),
+                  _isPublicVideo == false
+                      ? Padding(
+                          padding: const EdgeInsets.all(30.0),
+                          child: TextField(
+                            decoration: const InputDecoration(
+                              border: OutlineInputBorder(),
+                              labelText: 'Enter a token',
+                            ),
+                            controller: _tokenTextEditingController,
+                            onSubmitted: (value) async {
+                              if (_controller == null) {
+                                setState(() {
+                                  _controller = ApiVideoPlayerController(
+                                    videoOptions: VideoOptions(
+                                        videoId: _textEditingController.text,
+                                        token: value),
+                                  );
+                                });
+                              } else {
+                                _controller?.setVideoOptions(VideoOptions(
+                                    videoId: _textEditingController.text,
+                                    token: value));
+                              }
+                            },
+                          ),
+                        )
+                      : const SizedBox.shrink(),
```

Comment:
It is too complex for a sample. Just a second `TextField` that say `leave empty for public video` would have been enough